Userland/Libraries/LibArchive/Tar.h

```diff
@@ -51,14 +51,15 @@ constexpr StringView posix1_tar_magic = ""sv;   // POSIX.1-1988 format magic
 constexpr StringView posix1_tar_version = ""sv; // POSIX.1-1988 format version
 
 template<size_t N>
-static size_t get_field_as_integral(char const (&field)[N])
+static ErrorOr<unsigned> get_field_as_integral(char const (&field)[N])
```

Comment:
This should stay as a `size_t`.

Reply:
No. not really. You should be able to fix that though by doing a little gymnastic using `TRY` instead of returning the compound type as-is:

```cpp
ErrorOr<unsigned> checksum() const { return TRY(get_field_as_integral(m_checksum)); }
```

In this case, the `TRY` will "split up" the `ErrorOr<>` and will cause it to handle value and error separately, so it's able to pack either into a new `ErrorOr<>` that fits the typing requirements.